pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -162,6 +175,20 @@ func applyWebhookPolicy(w *autoscalingv1.WebhookPolicy, f *agonesv1.Fleet) (repl
 		return 0, false, err
 	}
 
+	defer func() {
```

Comment:
Rather than this fancy defer stuff 😄 should we create a new function called `applyWebhookPolicyWithFallback()`

That replaces the call above to:

```go
	case autoscalingv1.WebhookPolicyType:
		return applyWebhookPolicyWithFallback(pol.Webhook, f, gameServerLister, nodeCounts)
```

Which can call `applyWebhookPolicy` and then does the appropriate fallback handling with error management, without the complexity of working out what is happening in a `defer` statement? 😄